sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -114,6 +114,44 @@ const addRegistrationToDoc = (doc, registrations) => {
   }
 };
 
+const addReportUUIDToRegistration = (doc, registrations, callback) => {
+  if (registrations.length) {
+    var validRegistration;
+    var visitReportedDate = doc.reported_date;
+
+    outerloop: {
```

Comment:
Labels are not considered good form unless absolutely required, for similar reasons that `goto` in other languages is considered a bad idea.

In this and most instances, it's clearly a cry for this functionality to be extracted out into its own function.

This could be as simple as taking the code contained in `outerloop` and putting it in a function, replacing `break outerloop;` with `return`.

In fact, if we look at what `addReportUUIDToRegistration` is doing, we see it's doing two things:
 - Introspecting all of the registrations given and finding the "first" one that is valid to update
 - If it finds one, storing that registration change

Those sound very clearly like two separate functions to me.